Feature/enum impl

gaikoku-jin opened the pull request:
implementing enums instead of ints to cover mode of program operation as well as off/working/holiday status

Commit:
enums stub (in progress)

Commit:
main refurbished to operate on enum instead of int modes

Commit:
function introduced to obtain particular enum values (for OperationMode) relating to numerical values

Commit:
modifier refurbished to operate on enum instead of int modes

Commit:
updating project data

Commit:
enum implementation regarding to mapping working/off/holiday

Commit:
solving version conflict

Commit:
amending workspace.xml setting file

Commit:
expanding gitignore

Commit:
correcting allowanceValidator by estimation of off length

gaikoku-jin committed:
Merge branch 'develop' into feature/enum_impl

wasper1 commented on Urlopik/.idea/dictionaries/Marcin.xml:
```diff
@@ -0,0 +1,3 @@
+<component name="ProjectDictionaryState">
```
zignoruj i wyrzuc z commita cały folder .idea - to sa pliki konfiguracyjne narzędzia - nie trzyma się tego w repozytorium

wasper1 commented on Urlopik/Urlopik.iml:
```diff
@@ -1,7 +1,6 @@
 <?xml version="1.0" encoding="UTF-8"?>
```
Ten plik tez do wywalenia i zignorowania

wasper1 commented on Urlopik/.gitignore:
```diff
@@ -3,10 +3,10 @@
 # Created by https://www.gitignore.io/api/intellij
 
```
Zignoruj wszystkie skompilowane pliki (.class) i wyrzuć je z commita

wasper1 commented on Urlopik/src/main/Counter.java:
```diff
@@ -11,22 +12,22 @@
 	
 			
-	public Counter(Map<LocalDate, Integer> yearMap, int total) {
+	public Counter(Map<LocalDate, Enum> yearMap, int total) {
```
Map<LocalDate, Enum> zamień na Map<LocalDate, OffMode> to Enum jest zbyt generyczne i nie zwiększa czytelności :) a tak będzie wiadomo że mapujemy date na OffMode

wasper1 commented on Urlopik/src/main/Enums.java:
```diff
@@ -0,0 +1,48 @@
+
+public class Enums {
+
+    public enum OffMode {
```
Każdy enum do osobnej klasy

wasper1 commented on Urlopik/src/main/Modifier.java:
```diff
@@ -5,42 +5,43 @@
-		
 
-		if ((mode==1) && (remaining==0)){
+		if ((mode == OperationMode.SET) && (remaining == 0)){
```
podwójne nawiasowanie jest zbędne

wasper1 commented on Urlopik/src/main/Modifier.java:
```diff
@@ -49,36 +50,36 @@ else if ((mode==2) && (planned>0)){
 			}
-		
+
 		} while (validD==false || validA==false || trigger==true);
```
Z reguły jak jest dużo parametrów w jednym if'ie to robi sie  z tego metode typu shouldDoSth(parametr). Potem taką metodę mozna sobie przetestowac osobno

Commit:
.gitignore update

Commit:
Merge branch 'feature/enum_impl' of https://github.com/M4rc1N3k/Urlopik into feature/enum_impl

Commit:
removing trash from repo

Commit:
removing *.class files from repo

Commit:
separating Off- & OperationMode enums to classes

- as stated above
- providing necessary related changes in other classes

Commit:
extracting multiple alternative to function

Commit:
minor changes

Commit:
fixing allowanceValidator function

- incorporating adjacent operations related to this function
- goal: making the function compact for prospective testing

wasper1 commented on Urlopik/src/main/Modifier.java:
```diff
@@ -53,11 +53,9 @@ else if ((mode == OperationMode.CANCEL) && (planned > 0)){
-				}
-				else validA=true;
+
+				validA = allowanceValidator(mode, yearMap, start, end, remaining);			//checking if remaining allowance is larger than planned off days
```
dążymy do tego żeby kod był 'samokomentujący' = unikamy komantarzy i zamiast tego nazywamy funkcje tak żeby było wiadomo co robi jak przeczytasz jej nazwe :)

wasper1 commented on Urlopik/src/main/Modifier.java:
```diff
@@ -112,12 +110,20 @@ else if (en.getYear()>LocalDate.now().getYear())
+	private static Boolean allowanceValidator (OperationMode mode, Map<LocalDate, OffMode> yearMap, LocalDate start, LocalDate end, int remaining) {
+
+		if (mode == OperationMode.CANCEL)
+			return true;
```
unikaj braku {} jak ognia ! :dancing_women:

Commit:
final maintenance in the branch

- clearing comments
- changing variables' names to be self-descriptive
- putting {} brackets where necessary (if-else structures)
